In [1]:
import pandas as pd

In [2]:
campaign_data = {
    "Network": ["Meta Ads", "Google AC", "Applovin", "TikTok"],
    "Spend ($)": [2000, 1500, 1200, 800],
    "Installs": [1000, 800, 700, 600],
    "CPI ($)": [2.00, 1.88, 1.71, 1.33],
    "D1 Retention": [0.42, 0.38, 0.33, 0.40],
    "D3 Retention": [0.25, 0.22, 0.19,0.20],
    "D7 ROAS (%)": [35, 45, 30, 20]
}

In [3]:
df = pd.DataFrame(campaign_data)

df= df.set_index("Network") # I set the 'Network' column as an index to make the data easier to read.

df

Spend ($)  Installs  CPI ($)  D1 Retention  D3 Retention  \
Network                                                               
Meta Ads        2000      1000     2.00          0.42          0.25   
Google AC       1500       800     1.88          0.38          0.22   
Applovin        1200       700     1.71          0.33          0.19   
TikTok           800       600     1.33          0.40          0.20   

           D7 ROAS (%)  
Network                 
Meta Ads            35  
Google AC           45  
Applovin            30  
TikTok              20

##  PART 1 — Data Analysis

### Tasks : 

### 1. Calculate the Cost per Retained User (Day 3) for each network.

In [4]:
# Cost per Retained User (Day-3) = Spend / (Installs × D3_retention)

In [5]:
df['D3_CPRU ($)'] = df['Spend ($)'] / (df['Installs'] * df['D3 Retention'])
df

Spend ($)  Installs  CPI ($)  D1 Retention  D3 Retention  \
Network                                                               
Meta Ads        2000      1000     2.00          0.42          0.25   
Google AC       1500       800     1.88          0.38          0.22   
Applovin        1200       700     1.71          0.33          0.19   
TikTok           800       600     1.33          0.40          0.20   

           D7 ROAS (%)  D3_CPRU ($)  
Network                              
Meta Ads            35     8.000000  
Google AC           45     8.522727  
Applovin            30     9.022556  
TikTok              20     6.666667

### 2. Rank the networks based on cost efficiency and ROI potential.

In [6]:
# Cost Efficiency = > Lower is better
# ROI (Return on Investment) = Net Profit / Cost of Investment = > Higher is better

In [7]:
df_cost_efficiency = df.sort_values(by='D3_CPRU ($)', ascending=True)

print("Cost Efficiency Ranking")
print(df_cost_efficiency[['D3_CPRU ($)']])

Cost Efficiency Ranking
           D3_CPRU ($)
Network               
TikTok        6.666667
Meta Ads      8.000000
Google AC     8.522727
Applovin      9.022556


In [8]:
df_roi_potential = df.sort_values(by='D7 ROAS (%)', ascending=False)

print("ROI Potential Ranking")
print(df_roi_potential[['D7 ROAS (%)']])

ROI Potential Ranking
           D7 ROAS (%)
Network               
Google AC           45
Meta Ads            35
Applovin            30
TikTok              20


### 3. Based on the data, which network(s) would you scale next, pause, or re-test — and why?

In [9]:
# Average D3_CPRU and D7 ROAS

df_cost_efficiency[['D3_CPRU ($)']].mean() , df_roi_potential[['D7 ROAS (%)']].mean()

(D3_CPRU ($)    8.052988
 dtype: float64,
 D7 ROAS (%)    32.5
 dtype: float64)

- Google AC → Scale

  Why?: En yüksek D7 ROAS (45%) → ROI potansiyeli en yüksek. CPRU biraz fazla (≈$8.52). 


- Meta Ads → Scale (test-and-scale)

  Why?: Dengeli; D7 ROAS 35% (iyi) ve CPDRU ortalamaya yakın ($8.00).


- TikTok → Re-Test 

  Why?: En düşük CPDRU ($6.67) yani kullanıcı maliyeti düşük ama D7 ROAS çok düşük (20%) — kullanıcı kalitesi/monetizasyon sorunlu.


- Applovin → Pause

  Why?: Hem en yüksek CPRU (~$9.02) hem de düşük-orta ROAS (30%).


***Açıklama : Daha çok ortalamaları baz alarak karar oluşturturdum. Diğer taskda bu kararlar doğrultusunda performans geliştirme çalışması yapmayı planladım.

### 4. Identify 2–3 possible hypotheses to improve performance (e.g., creative direction, audience type, event optimization). 

- CPRU = Spend / (Installs × Retention)
- ROAS = (Revenue from ad campaign / cost of ad campaign ) * 100
- CPI = Total ad spend / Total installs

In [10]:
# The game is monetized with interstitial ads and rewarded videos, with a small in-app purchase (IAP) share

##### Audience Type Hypothesis (TikTok)
 
TikTok has the cheapest CPRU ($6.67) but the lowest D7 ROAS (20%).

This suggests that the platform attracts very low-cost but "low-quality" users.

Hypothesis: If we narrow the target audience using high-retention peers or specific interest groups (puzzle games), TikTok campaigns will attract higher LTV users, increasing D7 ROAS while maintaining a low CPRU.



##### Creative Direction Hypothesis (Meta)

Hypothesis: If we switch the campaign optimization event from install to a deeper value-based event (in-app purchase), Meta’s delivery algorithm will prioritize higher-quality users with stronger monetization potential.As a result, we expect an increase in D7 ROAS and lower Cost per Retained User (CPRU), even if CPI slightly increases.


##### Audience Optimization Hypothesis (Meta)

Meta has a good D7 ROAS, but its CPRU is slightly high. High-quality but expensive users. This may indicate that campaigns are still focused on existing audiences and the system is not sufficiently exploring less expensive users.

Hypothesis: If the campaign target audience is optimized and expanded to broader or more cost-effective lookalike/interest audiences, the system can achieve cheaper installs and the CPI will decrease.

## PART 2 — Creative Performance Deep Dive

### Tasks:

In [11]:
performance_data = {
    'Creative': ['A – Funny Fails', 'B – Satisfying Merge', 'C – Challenge Level'],
    'CTR (%)': [3.8, 2.2, 4.5],
    'CVR (%)': [40, 45, 35],
    'CPI ($)': [2.10, 1.80, 1.60],
    'D1 Retention': ['43%', '41%', '35%'],
    'D7 ROAS (%)': [32, 38, 25]
}

In [12]:
df_meta = pd.DataFrame(performance_data)

df_meta= df_meta.set_index("Creative")

df_meta

CTR (%)  CVR (%)  CPI ($) D1 Retention  D7 ROAS (%)
Creative                                                                 
A – Funny Fails           3.8       40      2.1          43%           32
B – Satisfying Merge      2.2       45      1.8          41%           38
C – Challenge Level       4.5       35      1.6          35%           25

##### Analyze the relationship between CTR, CVR, CPI, and ROAS.

In [13]:
# CTR (Click-through rate) => (Number of clicks / number of impressions) * 100
# CVR (Conversion rate) => (conversions / total visitors) * 100

Install rate per impression ≈ CTR × CVR

- A: 3.8% × 40% = 0.038 × 0.40 = 0.0152 = 1.52% (installs per impression)

- B: 2.2% × 45% = 0.022 × 0.45 = 0.0099 = 0.99%

- C: 4.5% × 35% = 0.045 × 0.35 = 0.01575 = 1.575%

C and A have the highest display→install efficiency while B the lowest.

If we divide the D7 ROAS percentage by the CPI, we have a comparison of which creative gives the better D7 return per spend.

- A: 32 / 2.10 = ≈ 15.24

- B: 38 / 1.80 = ≈ 21.11

- C: 25 / 1.60 = ≈ 15.63

B has the highest D7 return-per-spend density (best ROI). A and C are close, but B is clearly ahead.

C: Low cost and high traffic (scalability advantage), but low user quality and revenue.

B: Low CTR but high CVR and highest D7 ROAS—attracts more quality users; high cost-effectiveness.

A: Good CTR and very good D1 retention, but moderate D7 ROAS.

##### Which creative would you continue, pause, or iterate on?

B – Satisfying Merge => Scale

Why: Highest D7 ROAS/CPI, best D7 ROAS (38%). High CVR.

C – Challenge Level	=> Iterate

Why: Lowest CPI and highest CTR → scalable, but poor D1 retention and D7 ROAS. This means your creative isn't attracting the right audience.

A – Funny Fails	=> Iterate

Why: Balanced metrics, D1 retention is highest but D7 ROAS is average. A is a good creative.

Priority B > A > C

#####  Suggest two data-backed creative iteration ideas (e.g., new opening hook, visual tone, or CTA variation)

Iteration 1 — Expectation alignment (C için)

C yüksek CTR ama düşük retention/ROAS ⇒ reklam kullanıcıya yanlış beklenti veriyor olabilir

Fikir: Reklamın ilk 2–3 saniyesini değiştirmek: “Gerçek oyun akışı” veya “gerçek ödül sahneleri” gösteren 2 varyant üret. Bir varyantta oyun içi ödül/ödüllendirme (reward hint) vurgulanacak; diğer varyantta sosyal/rekabet yönü azaltılacak ve kolay başarı hissi vurgulanacak.

Amacımız D1 retention ve D7 ROAS yüzdesini arttırmak. Ayrıca CPI kabul edilebilir sınırda kalmalı.

Test setup: A/B test (C_original vs C_variant1 vs C_variant2). Her varyant için %50 - %50 ayırım yapıp sonrasında istatistiksel test

istatistiksel test: D7 ROAS'ta % istatiksel olarak anlamlılık (p < 0.05) kontrolü. (z-test daha mantıklı olur.)


Iteration 2 — CTA (A için)

A iyi retention ama B daha iyi monetizasyon. A’nın retention’ını monetize etmeye çalışabiliriz.

Hedefimiz A için D7 ROAS ve ARPU arttırmak.

(A için): Reklam sonuna kısa bir CTA/teaser eklenebilir — “İlk 24 saatte özel ödül” veya “2. seviyede bonus” gibi gerçek oyun içi değer vurgusu. Bu, IAP veya rewarded video (özellikle rewarded video) etkileşimini artırabilir.

Test setup: Her kreatif için tek değişkenli A/B test.

## References

- https://tr-tr.facebook.com/business/help/139628083350822
- https://ads.tiktok.com/help/article/creative-center?lang=tr
- https://www.appsflyer.com/glossary/interstitials/
- https://metalla.digital/meta-ad-hooks-that-drive-conversions-in-2025/
- https://www.ideasoft.com.tr/call-to-action-cta-nedir/
- https://segwise.ai/blog/optimization-techniques-meta-ad-campaigns-mobile-gaming
- https://www.facebook.com/business/ads-guide/update/video/audience-network-native/app-installs